collections.js

```diff
@@ -7,7 +7,7 @@ global.List = require("./list");
 global.Set = require("./set");
 global.Map = require("./map");
 global.MultiMap = require("./multi-map");
-global.WeakMap = require("weak-map");
+global.WeakMap = window.WeakMap || require("weak-map");
```

Comment:
Ok. To be clear, this does avoid executing the module but does not affect loading or parsing in production, and executing does essentially what you have unrolled but only once when the module is executed first. I am not convinced that this renders a real performance gain.